### PR TITLE
check state before deleting model or task

### PR DESCRIPTION
**Pull request by rbhavna**
Signed-off-by: Bhavana Goud Ramaram <rbhavna@amazon.com>

### Description
This PR check the condition for model state before deleting a model. If the model is in loaded/loading/partially loaded state, it throws exception and tells user to unload the model first. Previously, the delete API would delete the model irrespective of any model state which was no desirable.

Also checks condition for task state before deleting a task. If task is in running state, it throws an exception.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Zhangxunmt — APPROVED** [review]
LGTM. But to be on the safe side, are you still planning to check the model and task status in the cache memory to make sure they are not deleted? You can add it in a separate PR. https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/model/MLModelCache.java, and,  https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/task/MLTaskCache.java.

**Comment by rbhavna** [comment]
> LGTM. But to be on the safe side, are you still planning to check the model and task status in the cache memory to make sure they are not deleted? You can add it in a separate PR. https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/model/MLModelCache.java, and, https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/task/MLTaskCache.java.

Yes, I think we might have to do it.

File: plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java [hunk 1]
```diff
@@ -44,36 +55,66 @@ public class DeleteModelTransportAction extends HandledTransportAction<ActionReq
...
+                            });
+                        }
+                    } catch (Exception e) {
+                        log.error("Failed to parse ml model" + r.getId(), e);
+                        actionListener.onFailure(e);
                     }
-                    actionListener.onFailure(e);
                 }
-            });
+            }, e -> { actionListener.onFailure(new MLResourceNotFoundException("Fail to find model")); }), () -> context.restore()));
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java:117]: From line 81, we are going to restore context before running listener to delete model. Will it fail on security enabled cluster?

**Review by ylwu-amzn — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java (hunk 1 above)
Comment (rbhavna) [reply plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java:117]: Tested it on security enabled cluster and it works. Adding test results here.
Screenshot 2023-02-20 at 1 13 16 PM

**Pull request merged**